analyzers/src/SonarAnalyzer.Common/Helpers/ParameterLoader.cs

```diff
@@ -39,84 +36,32 @@ internal static class ParameterLoader
          * - diffing the contents of the configuration file
          * - associating the file with a unique identifier for the build project
          */
-        internal static void SetParameterValues(ParametrizedDiagnosticAnalyzer parameteredAnalyzer,
-            AnalyzerOptions options)
+        internal static void SetParameterValues(ParametrizedDiagnosticAnalyzer parameteredAnalyzer, SonarLintXmlReader sonarLintReader)
         {
-            var sonarLintXml = options.SonarLintXml();
-            if (sonarLintXml == null)
-            {
-                return;
-            }
-
-            var parameters = ParseParameters(sonarLintXml);
-            if (parameters.IsEmpty)
+            if (!sonarLintReader.ParametrizedRules.Any())
             {
                 return;
             }
 
             var propertyParameterPairs = parameteredAnalyzer.GetType()
                 .GetRuntimeProperties()
-                .Select(p => new { Property = p, Descriptor = p.GetCustomAttributes<RuleParameterAttribute>().SingleOrDefault() })
-                .Where(p => p.Descriptor != null);
+                .Select(x => new { Property = x, Descriptor = x.GetCustomAttributes<RuleParameterAttribute>().SingleOrDefault() })
+                .Where(x => x.Descriptor is { });
 
             var ids = new HashSet<string>(parameteredAnalyzer.SupportedDiagnostics.Select(diagnostic => diagnostic.Id));
             foreach (var propertyParameterPair in propertyParameterPairs)
             {
-                var parameter = parameters
-                    .FirstOrDefault(p => ids.Contains(p.RuleId));
+                var parameter = sonarLintReader.ParametrizedRules.FirstOrDefault(x => ids.Contains(x.Key));
 
-                var parameterValue = parameter?.ParameterValues
-                    .FirstOrDefault(pv => pv.ParameterKey == propertyParameterPair.Descriptor.Key);
+                var parameterValue = parameter?.Parameters.FirstOrDefault(x => x.Key == propertyParameterPair.Descriptor.Key);
```

Comment:
Empty line 55 doesn't help with anything. Same for 57.

If we would need semicolon-newline-newline as a command separator to make language readable, we would need a different language :)